test/utils/snap.go

```diff
@@ -69,6 +69,14 @@ func SnapVersion(t *testing.T, name string) string {
 	return strings.TrimSpace(out)
 }
 
+func SnapRevision(t *testing.T, service string) string {
+	out, _ := exec(t, fmt.Sprintf(
+		"snap list | grep %s | awk '{print $3}'",
+		service,
```

Comment:
```suggestion
func SnapRevision(t *testing.T, name string) string {
	out, _ := exec(t, fmt.Sprintf(
		"snap list %s | awk 'NR==2 {print $3}'", //  extract the 3rd column of 2nd line
		name,
```
This will query one snap and get the 3rd value from second line. First line is the header.
This is better than grep because that will fail when we have multiple versions of a snap installed.


```
$ snap install edgex-device-mqtt --stable
edgex-device-mqtt 2.2.0 from Canonical✓ installed
$ snap install edgex-device-mqtt_2 --edge
edgex-device-mqtt_2 (edge) 2.3.0-dev.10 from Canonical✓ installed

$ snap list | grep edgex-device-mqtt
edgex-device-mqtt               2.2.0                       2369   latest/stable    canonical**          -
edgex-device-mqtt_2             2.3.0-dev.10                2530   latest/edge      canonical**          -

# this is not what you want:
$ snap list | grep edgex-device-mqtt | awk '{print $3}'
2369
2530

# this works:
$ snap list edgex-device-mqtt | awk 'NR==2 {print $2}'
2.2.0
```

We could use the same command for SnapVersion and get the second column.